third_party/terraform/utils/bootstrap_utils_test.go

```diff
@@ -18,23 +18,28 @@ type bootstrappedKMS struct {
 	*cloudkms.CryptoKey
 }
 
-// BootstrapKMSKey returns a KMS key in the "global" location.
-// See BootstrapKMSKeyInLocation.
+// BootstrapKMSKeyWithPurposeAlgorithm returns a KMS key in the "global" location.
+// See BootstrapKMSKeyWithPurposeAlgorithmInLocation.
+func BootstrapKMSKeyWithPurposeAlgorithm(t *testing.T, purpose, algorithm string) bootstrappedKMS {
+	return BootstrapKMSKeyWithPurposeAlgorithmInLocation(t, purpose, algorithm, "global")
+}
+
 func BootstrapKMSKey(t *testing.T) bootstrappedKMS {
 	return BootstrapKMSKeyInLocation(t, "global")
 }
 
 /**
-* BootstrapKMSKeyWithLocation will return a KMS key in a particular location
-* that can be used in tests that are testing KMS integration with other resources.
+* BootstrapKMSKeyWithPurposeAlgorithmWithLocation will return a KMS key in a
+* particular location with the given purpose and alogrithm that can be used
+* in tests that are testing KMS integration with other resources.
 *
 * This will either return an existing key or create one if it hasn't been created
```

Comment:
This is currently not checking purpose and algorithm before checking for existing keys. I still need to adapt this to return the right type of key depending on the arguments passed.

Reply:
Fixed. purpose specific keys are no longer share a name.